### ISSUE TITLE
Mining Fatigue potion effect conflicts with other plugins

### ISSUE THREAD

**Issue by Rumsfield** [Athlaeos/ValhallaMMO#10]
For context, I have not used ValhallaMMO, but another user brought this issue to my attention. I am the author of another Spigot plugin which the user is trying to use along with ValhallaMMO.

In the other plugin, chunks of land belong to claimed territories. When players enter or exit these claimed territories, the Mining Fatigue potion effect can be applied or removed from the player.

The problem is that there is apparently a mechanic in ValhallaMMO which relies on the level of the Mining Fatigue potion effect for various stats. When the other plugin removes this potion effect, it breaks features in ValhallaMMO. I'd like to understand how ValhallaMMO uses potion effects, and how my plugin can potentially integrate to work around these mechanics.

**Comment by Athlaeos** [comment]
Sorry for the lack of response, I wasn't notified for some reason

Mining Fatigue is required to make the client stop seeing block cracks and to make players sufficiently unable to break blocks normally. Valhalla has its own block breaking system and so it handles block breaking and speeds and crack animations itself. In 1.20.5 and above this effect is no longer required as it can now utilize a negative mining speed attribute instead of potion effect. 

I would check if a player's effect duration isn't ludicrously high, because the fatigue effect applied by Valhalla has the integer limit as duration. You could also make a blacklist of potion effects to ignore so that mining fatigue can be kept.
I'm not sure if this is sufficient information as "how ValhallaMMO uses potion effects" is quite vague and can mean anything, but if you want to see for yourself how the effect is applied and handled you can look here https://github.com/Athlaeos/ValhallaMMO/blob/master/core/src/main/java/me/athlaeos/valhallammo/listeners/CustomBreakSpeedListener.java

**Comment by NoPing420** [comment]
I am having a similar issue but I'm wondering why every player that joins my server has such a ridiculously high number of mining fatigue is it supposed to be like this?

**Comment by Athlaeos** [comment]
Yes, it's intentional. For 1.20.4 and below this is the only way to allow the custom mining system to function. If it's too disruptive, you can disable the system in config.yml
